bugbot/rules/perfalert_resolved_regression.py

```diff
@@ -0,0 +1,190 @@
+                bug_info["nickname"] = user_emails_to_names[bug_info["status_author"]]
+
+    def comment_handler(self, bug, bug_id, bugs):
+        # Gather all comments to match them with the history after
+        self._bug_comments[bug_id] = bug["comments"]
+
+    def history_handler(self, bug):
+        bug_info = self._bug_history.setdefault(str(bug["id"]), {})
+
+        # Get the last resolution change that was made in this bug
+        for change in bug["history"][::-1]:
+            # Get the most recent resolution change first, this is because
+            # it could have changed since the status was changed and by who
+            if not bug_info.get("resolution"):
+                for specific_change in change["changes"]:
+                    if specific_change["field_name"] == "resolution":
+                        bug_info["resolution"] = specific_change["added"]
+                        bug_info["resolution_previous"] = (
+                            specific_change["removed"].strip() or "---"
+                        )
+                        bug_info["resolution_time"] = change["when"]
+                        break
+
+            if bug_info.get("resolution"):
+                # Find the status that the bug was resolved to, and by who
+                for specific_change in change["changes"]:
+                    if specific_change["field_name"] == "status" and specific_change[
+                        "added"
+                    ] in ("RESOLVED", "REOPENED"):
+                        bug_info["status"] = specific_change["added"]
+                        bug_info["status_author"] = change["who"]
+                        bug_info["status_time"] = change["when"]
+                        break
+
+            if bug_info.get("status"):
+                break
+
+    def gather_bugs_info(self, bugs):
+        Bugzilla(
+            bugids=self.get_list_bugs(bugs),
+            historyhandler=self.history_handler,
+            commenthandler=self.comment_handler,
+            commentdata=bugs,
+            comment_include_fields=["text", "creation_time", "author"],
+        ).get_data().wait()
+
+        # Match the history with comments to get resolution reasons
+        self.get_resolution_comments(bugs)
+
+        # Get info for needinfos
+        self.get_needinfo_nicks()
+
+    def set_autofix(self, bugs):
+        for bug_id, bug_info in self.extra_ni.items():
+            self.add_auto_ni(
+                bug_id,
+                {
+                    "mail": bug_info["status_author"],
+                    "nickname": (
+                        (":" + bug_info["nickname"])
+                        if "nickname" in bug_info
+                        else bug_info["status_author"]
+                    ),
+                },
+            )
+
+    def get_bugs(self, *args, **kwargs):
+        bugs = super().get_bugs(*args, **kwargs)
+        self.gather_bugs_info(bugs)
+        self.set_autofix(bugs)
+        return bugs
```

Comment:
We could drop this method and handle the bugs individually using the `handle_bug()` method.

Example: https://github.com/mozilla/bugbot/blob/a8b33bedcf4c7cf0e0deb5448b2748912ffe9ea7/bugbot/rules/not_landed.py#L49-L67

Reply:
I don't think we can drop this completely since I still need to get the nick for whoever resolved the bug. It's not always the person that is assigned to the bug, and I don't see another way of getting the nick for that person. I can only get the email of the person from the comments.

Reply:
Oh, you. This is correct. 

It would be nicer if the nicknames were retrieved automatically when not provided. I will file an issue to improve it in the future.